docs/ref/forms/api.txt

```diff
@@ -260,9 +262,30 @@ precedence::
 
 .. method:: Form.get_initial_for_field(field, field_name)
 
-Use :meth:`~Form.get_initial_for_field()` to retrieve initial data for a form
-field. It retrieves data from :attr:`Form.initial` and :attr:`Field.initial`,
-in that order, and evaluates any callable initial values.
+Returns the initial data for a form field. The data is retrieved from
+:attr:`Form.initial` if present, otherwise :attr:`Field.initial`, and
+evaluated if a callable.
+
+It is recommended to use :attr:`BoundField.initial` over
+:meth:`~Form.get_initial_for_field()` because ``BoundField.initial`` has a
+simpler interface. Also, unlike :meth:`~Form.get_initial_for_field()`,
+:attr:`BoundField.initial` caches its values. This is useful especially when
+dealing with callables whose return values can change (e.g. ``datetime.now`` or
+``uuid.uuid4``)::
+
+        >>> import uuid
+        >>> class UUIDCommentForm(CommentForm):
+        ...     identifier = forms.UUIDField(initial=uuid.uuid4)
+        >>> f = UUIDCommentForm()
+        >>> print(f.get_initial_for_field(f.fields['identifier'], 'identifier'))
+        972ca9e4-7bfe-4f5b-af7d-07b3aa306334
+        >>> print(f.get_initial_for_field(f.fields['identifier'], 'identifier'))
+        69699b65-5679-47fa-a518-3c9b872e5be8
+        # Using BoundField.initial, for comparison
+        >>> print(f['identifier'].initial)
+        9c50c0f6-966f-475e-9e1f-fb591e5dadaf
+        >>> print(f['identifier'].initial)
+        9c50c0f6-966f-475e-9e1f-fb591e5dadaf
```

Comment:
It's over-indented:
 ```suggestion
    >>> import uuid
    >>> class UUIDCommentForm(CommentForm):
    ...     identifier = forms.UUIDField(initial=uuid.uuid4)
    >>> f = UUIDCommentForm()
    >>> print(f.get_initial_for_field(f.fields['identifier'], 'identifier'))
    972ca9e4-7bfe-4f5b-af7d-07b3aa306334
    >>> print(f.get_initial_for_field(f.fields['identifier'], 'identifier'))
    69699b65-5679-47fa-a518-3c9b872e5be8
    # Using BoundField.initial, for comparison
    >>> print(f['identifier'].initial)
    9c50c0f6-966f-475e-9e1f-fb591e5dadaf
    >>> print(f['identifier'].initial)
    9c50c0f6-966f-475e-9e1f-fb591e5dadaf
```

Reply:
I would also remove `print()` from both examples. It's worth showing that `initial` returns `UUID` and `datetime` objects.
```python
        >>> f['identifier'].initial
        UUID('9c50c0f6-966f-475e-9e1f-fb591e5dadaf')
```